rfc/rfc-0018-workload-report-artifact-tree.md

```diff
@@ -1,3 +1,202 @@
+Because controllers continuously reconcile their objects towards a desired state. A supply chain is, therefore, able to
+choreograph an otherwise set of independent objects (and their controllers).
+
+A workload acts as input into this supply chain seeding initial values into an object to make the first controller do
+work. The supply chain manages this workload input data and outputs from each object as it propagates through the
+supply chain.
+
+It is important to note however, that a workload can have multiple inputs and input into several objects in the supply
+chain all at the same time. As a result when a workload changes, it may cause several controllers to do work
+simultaneously.
+
+It is also important to note that a controller may also do work outside of the work that the supply chain is
+choreographing. Kpack images, for example, are often choreographed as part of a supply chain. But the kpack controller
+may also do work in response to a base OS image update. Nothing to do with the supply chain but impacting it
+none-the-less. As a result the supply chain may be triggered part way through by this input.
+
+### Problem
+
+Whilst, on the one hand, the behavior described above is virtuous and generally beneficial to automated outer loop
+workflows. On the other hand, this behavior can be problematic for more user-centric, imperative workflows, such as
+those often found in the inner loop. Those of debugging and live update.
+
+### Providing a real world example.
+
+Given a supply chain that has several inputs, let’s say source code image and a debug flag. When a developer applies a
+workload after changing their source code and turning on debug. Several services in the supply chain may trigger at the
+same time and these inputs will traverse through the supply chain at different times. But the developer, initiating the
+debug session will want to wait for that specific source code change to arrive “in cluster” in an image prepped for
+debugging before attaching their debugger. So, it is important to know when both inputs have fully traversed the supply
+chain.
+
+## Detailed Explanation
+
+Each workload/Deliverable will include in their status an artifacts field:
+```yaml
+status:
+  artifacts:
+    # oneOf(source,image,config, object)
+    - source:
+        # the sha256 of the ordered JSON of all other non-from fields
+        id: <SHA:string>
+        # exposed fields - in this case url and revision
+        uri: <:string>
+        revision: <:string>
+        # the object which produced this artifact
+        resource:
+          # name of the resource in the supply chain
+          resource-name: <:string>
+          # GVK of the resource
+          kind: <:string>
+          apiVersion: <:string>
+          # name of the resource on the cluster
+          name: <:string>
+          # namespace of the resource
+          namespace: <:string>
+          # resource version of the object
+          resourceVersion: <:string>
```

Comment:
Could this be an [`ObjectReference`](https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/object-reference/) or [`TypedLocalObjectReference`](https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/typed-local-object-reference)?